components/brave_shields/browser/brave_shields_util.cc

```diff
@@ -335,12 +355,21 @@ bool AllowReferrers(HostContentSettingsMap* map, const GURL& url) {
 void SetFingerprintingControlType(HostContentSettingsMap* map,
                                   ControlType type,
                                   const GURL& url,
-                                  PrefService* local_state) {
+                                  PrefService* local_state,
+                                  PrefService* profile_state) {
```

Comment:
or vice versa, erase local state and use only prefs

Reply:
Shields settings are per profile, so it may be best to store global/domain-specific setting counts separately. As a result, the info from the last used profile will be transmitted at P3A report time.

Reply:
The only shields P3A function that uses local_state is `MaybeRecordShieldsUsageP3A`, which was present before this PR. Perhaps it would be best to refactor that function to use profile prefs in a separate issue/PR